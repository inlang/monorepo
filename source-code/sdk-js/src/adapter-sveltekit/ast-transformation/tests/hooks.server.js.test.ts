import { describe, it } from 'vitest'
import {transformHooksServerJs, wrapHooksServerJs, createHooksServerJs } from '../transforms/hooks.server.js.js'
import type {TransformConfig} from "../config.js";

const baseConfig:TransformConfig = {
    isStatic: false,
    languageInUrl: false,
    srcFolder: '',
    rootRoutesFolder: '',
    hasAlreadyBeenInitialized: false,
}

describe('transform hooks.server.js', () => {
    it.concurrent('transformHooksServerJs returns new file if no code provided', async ({ expect }) => {
        const code = transformHooksServerJs(baseConfig, '')
        expect(code).toContain('getLanguage')
    })

    // todo: cover with real test once this implemented
    it.concurrent('wrapHooksServerJs', async ({ expect }) => {
        expect(() => wrapHooksServerJs(baseConfig, 'anything')).toThrowError('currently not supported')
    })

    it.concurrent('createHooksServerJs languageInUrl and isStatic', async ({ expect }) => {
        const config:TransformConfig = {
            ...baseConfig,
            languageInUrl: true,
            isStatic: true,
        }
        const code = createHooksServerJs(config)
        expect(code).toMatchSnapshot()
    })

    it.concurrent('createHooksServerJs isStatic', async ({ expect }) => {
        const config:TransformConfig = {
            ...baseConfig,
            isStatic: true,
        }
        const code = createHooksServerJs(config)
        expect(code).toMatchSnapshot()
    })
})