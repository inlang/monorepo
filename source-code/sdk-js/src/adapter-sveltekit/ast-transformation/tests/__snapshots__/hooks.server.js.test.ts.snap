// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`transform hooks.server.js > createHooksServerJs isStatic 1`] = `
"
import { initHandleWrapper } from \\"@inlang/sdk-js/adapter-sveltekit/server\\"
import { initAcceptLanguageHeaderDetector } from \\"@inlang/sdk-js/detectors/server\\"
import { redirect } from \\"@sveltejs/kit\\"
import { replaceLanguageInUrl } from \\"@inlang/sdk-js/adapter-sveltekit/shared\\"

export const handle = initHandleWrapper({
getLanguage: () => undefined,

}).wrap(async ({ event, resolve }) => resolve(event))
"
`;
